agent/eventhandler/handler.go

```diff
@@ -28,6 +28,8 @@ var statesaver statemanager.Saver = statemanager.NewNoopStateManager()
 
 func HandleEngineEvents(taskEngine engine.TaskEngine, client api.ECSClient, saver statemanager.Saver) {
 	statesaver = saver
+	handler := NewTaskHandler()
```

Comment:
it's intentional. there is a clear division of labor between the two and i could rename this to something more descriptive. maybe "taskEventHandler"?